akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpHeaderParser.scala

```diff
@@ -544,8 +553,21 @@ private[http] object HttpHeaderParser {
                 case -1 ⇒ if (sb != null) sb.append(c).append(byteChar(input, ix + 1)).append(byteChar(input, ix + 2)).append(byteChar(input, ix + 3)) else null
                 case cc ⇒ appended2(cc)
               }
-            } else fail(s"Illegal character '${escape(c)}' in header value")
-          scanHeaderValue(hhp, input, start, limit)(nsb, nix)
+            } else {
+              mode match {
+                case ParserSettings.IllegalResponseHeaderValueProcessingMode.Error ⇒
+                  fail(s"Illegal character '${escape(c)}' in header value")
+                case ParserSettings.IllegalResponseHeaderValueProcessingMode.Warn ⇒
+                  // ignore the illegal character and log a warning message
+                  log.warning(s"Illegal character '${escape(c)}' in header value")
+                  sb
+                case ParserSettings.IllegalResponseHeaderValueProcessingMode.Ignore ⇒
+                  // just ignore the illegal character
+                  sb
+              }
```

Comment:
I wonder if `IllegalResponseHeaderValueProcessingMode` should be `IllegalResponseHeaderValueHandler` of type `Char => Unit` so we can put this logic there and just call `handler.handle(c)` here.

Dunno how to pass `fail`/`escape` with the same perf. Do others think this change would be valuable?